Cargo.toml

```diff
@@ -19,13 +20,40 @@ include = [
     "/LICENSE-APACHE",
 ]
 
+[features]
+default = ["nb", "linux-embedded-hal"]
+nightly = ["async"]
+async = ["embedded-hal-async"]
+
 [dependencies]
 embedded-hal = "0.2.5"
-nb = "1"
+embedded-hal-async = { version = "0.1.0-alpha.1", optional = true }
+linux-embedded-hal = { version = "0.3", optional = true }
```

Comment:
This should only be a `dev-dependency`

Reply:
It's not possible to have an optional `dev-dependency`, so if I leave it there this crate is not buildable on macOS...

Reply:
We can do:
```toml
[target.'cfg(target_os = "linux")'.dev-dependencies]
linux-embedded-hal = "0.3"
```
The examples will still not build, though.

Reply:
If I do that, there's no way to indicate that the example requires it. At least with it as an optional dependency cargo will avoid building the example if the feature isn't enabled.

Reply:
Alright, the best I can come up with is to use `cfg` guards to make `linux.rs` into a noop.